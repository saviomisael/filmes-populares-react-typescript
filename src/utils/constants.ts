export const IMAGE_URL = process.env.REACT_APP_IMAGE_URL;
