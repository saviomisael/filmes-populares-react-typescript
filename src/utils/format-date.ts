export const formatDate = (date: Date) =>
  date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
