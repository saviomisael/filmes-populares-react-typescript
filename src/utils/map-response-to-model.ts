import { TvSerie } from 'models/TvSerie';
import { TvSerieResponse } from 'services/responses/TvSerieResponse';

const mapSingleResponseToTvSerieModel = (response: TvSerieResponse) =>
  new TvSerie(response.id, response.name, response.vote_average, response.poster_path);

export const mapResponseToTvSerieModel = (response: TvSerieResponse[]) =>
  response.map((x) => mapSingleResponseToTvSerieModel(x));
