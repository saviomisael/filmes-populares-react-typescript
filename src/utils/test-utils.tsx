import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
