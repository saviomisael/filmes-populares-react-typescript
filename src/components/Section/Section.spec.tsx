import { renderTheme } from 'utils/test-utils';
import { Section } from '.';

describe('<Section />', () => {
  it('should renders component with background orange', () => {
    const { container } = renderTheme(<Section bgColor="orange">Some text</Section>);

    expect(container.firstChild).toHaveStyleRule('background', '#FA9F42');
  });

  it('should renders component with background black', () => {
    const { container } = renderTheme(<Section bgColor="black">Some text</Section>);

    expect(container.firstChild).toHaveStyleRule('background', '#413B3B');
  });

  it('should matches with snapshot', () => {
    const { container } = renderTheme(<Section bgColor="orange">Some text</Section>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
