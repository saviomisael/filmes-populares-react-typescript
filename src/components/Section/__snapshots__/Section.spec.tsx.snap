// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should matches with snapshot 1`] = `
.c0 {
  padding: 1.5rem;
  background: #FA9F42;
}

<section
  class="c0"
>
  Some text
</section>
`;
